src/garage/experiment/meta_evaluator.py

```diff
@@ -0,0 +1,102 @@
+"""Evaluator which tests Meta-RL algorithms on test environments."""
+from dowel import tabular
+import numpy as np
+
+from garage.misc.tensor_utils import discount_cumsum
+from garage.sampler import LocalSampler
+
+
+class MetaEvaluator:
+    """Evaluates Meta-RL algorithms on test environments.
+
+    Args:
+        runner (garage.experiment.LocalRunner): Runner.
+        test_task_sampler (garage.experiment.TaskSampler): Sampler for test
+            tasks. To demonstrate the effectiveness of a meta-learning method,
+            these should be different from the training tasks.
+        max_path_length (int): Maximum path length used for evaluation
+            trajectories.
+        n_test_tasks (int or None): Number of test tasks to sample each time
+            evaluation is performed. Note that tasks are sampled "without
+            replacement". If None, is set to `test_task_sampler.n_tasks`.
+        prefix (str): Prefix to use when logging. Defaults to MetaTest. For
+            example, this results in logging the key 'MetaTest/SuccessRate'.
+            If not set to `MetaTest`, it should probably be set to `MetaTrain`.
+
+    """
+
+    def __init__(self,
+                 runner,
+                 *,
+                 test_task_sampler,
+                 max_path_length,
+                 n_test_tasks=None,
+                 prefix='MetaTest'):
+        self._test_task_sampler = test_task_sampler
+        if n_test_tasks is None:
+            n_test_tasks = test_task_sampler.n_tasks
+        self._n_test_tasks = n_test_tasks
+        self._test_sampler = runner.make_sampler(
+            LocalSampler, n_workers=1, max_path_length=max_path_length)
```

Comment:
should the number of workers be configurable? what if this is expensive?

Reply:
If it's too expensive (which seems fairly unlikely), I can add another worker and parallel sampling or something in a later PR.

Reply:
BTW, adding more workers here (with `LocalSampler`), wouldn't actually make anything faster.